core/src/main/java/com/datastrato/gravitino/connector/BaseCatalog.java

```diff
@@ -169,12 +184,57 @@ public CatalogOperations ops() {
     return ops;
   }
 
+  public AuthorizationPlugin getAuthorizationPlugin() {
+    if (authorization == null) {
+      synchronized (this) {
+        if (authorization == null) {
+          authorization = createAuthorizationPluginInstance();
+        }
+      }
+    }
+    return authorization.plugin();
+  }
+
+  private BaseAuthorization<?> createAuthorizationPluginInstance() {
+    String provider = entity.getProperties().get(AUTHORIZATION_PROVIDER);
```

Comment:
Can we use `PropertiesMetadata::getOrDefault` here?

Reply:
DONE.